Sources/SwiftInspectorVisitors/GenericRequirementVisitor.swift

```diff
@@ -0,0 +1,82 @@
+// Created by Dan Federman on 1/26/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import SwiftSyntax
+
+public final class GenericRequirementVisitor: SyntaxVisitor {
+
+  public var genericRequirements = [GenericRequirement]()
+
+  public override func visit(_ node: SameTypeRequirementSyntax) -> SyntaxVisitorContinueKind {
+    genericRequirements.append(GenericRequirement(from: node))
+    return .visitChildren
+  }
+
+  public override func visit(_ node: ConformanceRequirementSyntax) -> SyntaxVisitorContinueKind {
+    genericRequirements.append(GenericRequirement(from: node))
+    return .visitChildren
+  }
+
+}
+
+public struct GenericRequirement: Codable, Equatable {
+
+  // MARK: Lifecycle
+
+  public init(
+    leftType: String,
+    rightType: String,
+    relationship: Relationship)
+  {
+    self.leftType = leftType
+    self.rightType = rightType
+    self.relationship = relationship
+  }
+
+  init(from node: SameTypeRequirementSyntax) {
+    leftType = node.leftTypeIdentifier.description.trimmingCharacters(in: .whitespacesAndNewlines)
+    rightType = node.rightTypeIdentifier.description.trimmingCharacters(in: .whitespacesAndNewlines)
```

Comment:
I don't think we should use `description` here. IIUC `description` is just a representation of the instance (from CustomStringConvertible). The fact we're relying on that is a hack IMO that can break unexpectedly since there's no contract that states that this needs to be the same value as the source (e.g. we can imagine that this can change to be more user friendly by providing more information). 

I'd recommend iterating on the possible type of the identifier instead as how we do here https://github.com/fdiaz/SwiftInspector/blob/ae1ef42775efbdb85b9cbd5eb25aa49a5d169c34/Sources/SwiftInspectorAnalyzers/TypeConformanceAnalyzer.swift#L66

Reply:
Going to try this suggestion out. That said, the [doc comment on `description`](https://github.com/apple/swift-syntax/blob/577d3bbe3a4f39ba8843093f3bd1ae0b593da557/Sources/SwiftSyntax/Syntax.swift#L435) does make it seem like this is a reasonable API to use:
> A source-accurate description of this node.

Reply:
6be54ac145af0dcc587e65717e40f1c9248bc438

Reply:
and a bit of a painful followup 8db6d24fd31072097188d7d5c562afb9d4ba0c36

Reply:
I was wrong! See #68